src/go/rpk/pkg/cli/cmd/cluster/maintenance/enable.go

```diff
@@ -66,7 +70,48 @@ node exists that is already in maintenance mode then an error will be returned.
 
 			out.MaybeDie(err, "error enabling maintenance mode for node %d: %v", nodeId, err)
 			fmt.Printf("Successfully enabled maintenance mode for node %d\n", nodeId)
+
+			if !wait {
+				return
+			}
+
+			fmt.Println("Waiting for node to drain...")
+
+			var table *out.TabWriter
+			retries := 3
+			for {
+				b, err := client.Broker(nodeId)
+				if err == nil && b.Maintenance == nil {
+					err = fmt.Errorf("maintenance mode not supported. upgrade in progress?")
+					// since admin api client uses `sendAny` it is possible that
+					// we enabled maintenance mode and then started querying a
+					// broker that hadn't been upgraded yet.  with the retry,
+					// this is unlikely. and won't occur at all once all nodes
+					// are upgraded to v22.1.x.
+					//
+					// [[fallthrough]].
+				}
+				if err != nil {
+					if retries <= 0 {
+						out.Die("Error retrieving broker status: %v", err)
+					}
+					retries--
+					time.Sleep(time.Second * 2)
+					continue
+				}
+				retries = 3
+				if table == nil {
+					table = newMaintenanceReportTable()
+				}
+				addBrokerMaintenanceReport(table, b)
+				table.Flush()
+				if b.Maintenance.Draining && b.Maintenance.Finished {
```

Comment:
I worry a little about this waiting forever if someone disabled maintenance mode in the background, but I don't have a neat suggestion for making it safer.

Reply:
yeh that's a bit of an issue that could arise. I guess it'd be nice to have some sort of epoch/version number. presumably that would allow the waiter to reason about if draining had stopped due to another modification to the maintenance state. that passes the sniff test?